src/Features/Core/Portable/IncrementalCaches/SymbolTreeInfoIncrementalAnalyzerProvider.cs

```diff
@@ -234,6 +234,12 @@ private Task UpdateReferencesAync(Project project, CancellationToken cancellatio
             private async Task UpdateReferenceAsync(
                 Project project, PortableExecutableReference reference, CancellationToken cancellationToken)
             {
+                if (cancellationToken.IsCancellationRequested)
```

Comment:
I'm not seeing how this will improve things debugging roslyn.  We'll still be throwing cancellation exceptions in effectively every other codepath.  So how does this actually help things out?